### PR TITLE
[MI-3653]: Test cases of message sync of MS Teams

### PR DESCRIPTION
**Pull request by AayushChaudhary0001**
This PR consist of the following test cases:-

- Message sync(emoji, GiF, text message)
- Message deleted from Teams
- Undo deleted message on Teams
- Order of posts

### CODE REVIEW

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter the desired text message in the already linked MM channel by the account which is not already connected.
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:38]: is it necessary for the message to be in text format?

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+**Step 2**
+
+1. Connect the bot-account.
+2. Enter the desired text message in the already linked MM channel by the account which is not already connected.
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:43]: same

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+**Expected**
+
+The message should not be reflected on the linked Teams channel. After step 2, the message provided should be reflected by the bot account contaning the original user's name in the message in the linked channel in Teams.
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:47]:
```suggestion
The message should not be reflected on the linked Teams channel. After step 2, the message should be reflected by the bot account containing the original user's name in the message in the linked channel of Teams.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Order_of_posts.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+1. Enter multiple messages in the already linked MM channel or in DM/GM.
+
+**Step 2**
+
+1. Enter multiple messages in the already linked Teams channel or in DM/GM.
+
+**Expected**
+
+The messages should be synced from MM to Teams in the correct order as delivered already. After step 2, the messages should be synced from Teams to MM in the correct order as delivered already.
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Order_of_posts.md:46]:
```suggestion
The messages should be synced from MM to Teams in the correct order, as delivered already. After step 2, the messages should be synced from Teams to MM in the correct order, as delivered already.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Teams_to_MM.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter any text message, emoji, GiF, link, image in the already linked channel in Teams or in any DM/GM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Teams_to_MM.md:38]:
```suggestion
1. Enter any text message, emoji, GiF, link, image, etc. in the already linked channel in Teams or in any DM/GM.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Teams_to_MM.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+**Expected**
+
+The same message posted on MM should be visible on the linked channel in MM or DM/GM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Teams_to_MM.md:42]:
```suggestion
The same message posted on Teams should be visible in the linked channel in MM or DM/GM.
```

File: data/test-cases/plugins/MS Teams sync/Slash commands/Unlink_channels.md
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Slash commands/Unlink_channels.md:1]: why is this file present in this PR? It is already present here https://github.com/Brightscout/mattermost-test-management/pull/6/files#diff-b241f3e636cd6fc56b1c646261d70facff90d79ffe674fecbc6fae8704b769bc

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (AayushChaudhary0001) [reply data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:38]: Updated

File: data/test-cases/plugins/MS Teams sync/Message sync/Message_deleted.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+---
+# (Required) Ensure all values are filled up
+name: "Message deleted from Teams to MM"
+status: Active
+priority: Normal
+folder: Slash commands
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Message_deleted.md:6]: same. fix everywhere

File: data/test-cases/plugins/MS Teams sync/Message sync/Message_deleted.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Delete the desired message from the linked channel in Teams.
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Message_deleted.md:38]: should we add here as well `or in any DM/GM`

Comment (ayusht2810) [reply data/test-cases/plugins/MS Teams sync/Message sync/Message_deleted.md:38]: if updating, update everywhere

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+1. Enter the desired message in the already linked MM channel from the non-connected account.
+
+**Step 2**
+
+1. Connect the bot-account.
+2. Enter the desired message in the already linked MM channel by the account which is not already connected.
+
+**Expected**
+
+The message should not be reflected on the linked Teams channel. After step 2, the message should be reflected by the bot account the original user's name in the message in the linked channel in Teams.
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:47]:
```suggestion
The message should not be reflected on the linked Teams channel. After step 2, the message should be reflected by the bot account with the original user's name in the message in the linked channel in Teams.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
+---
+# (Required) Ensure all values are filled up
+name: "Message from non-connected user"
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:3]: should we also add a test case when a non-connected user messages in DM/GM or will that be present in a different test file?

File: data/test-cases/plugins/MS Teams sync/Message sync/Order_of_posts.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+
+1. Enter multiple messages in the already linked MM channel or in DM/GM.
+
+**Step 2**
+
+1. Enter multiple messages in the already linked Teams channel or in DM/GM.
+
+**Expected**
+
+The messages should be synced from MM to Teams in the correct order, as delivered already. After step 2, the messages should be synced from Teams to MM in the correct order, as delivered already.
```
Comment (ayusht2810) [inline data/test-cases/plugins/MS Teams sync/Message sync/Order_of_posts.md:46]: The handling of post order from MM to Teams is not done yet. Please remove this test case.

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md (hunk 10 above)
Comment (AayushChaudhary0001) [reply data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:3]: That will be covered under plugin configs

**Review by ayusht2810 — APPROVED** [review]

File: data/test-cases/plugins/MS Teams sync/Message sync/MM_to_Teams.md [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter any text message, emoji, GiF, link, image, etc. in the already linked channel in MM or in any DM/GM.
```
Comment (manojmalik20) [inline data/test-cases/plugins/MS Teams sync/Message sync/MM_to_Teams.md:38]:
```suggestion
1. Enter any text message, emoji, GIF, link, image, etc. in any already linked channel or DM/GM in MM.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/MM_to_Teams.md [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+**Expected**
+
+The same message posted on MM should be visible on the linked channel in Teams or DM/GM.
```
Comment (manojmalik20) [inline data/test-cases/plugins/MS Teams sync/Message sync/MM_to_Teams.md:42]: Replace all instances of "Teams" with "MS Teams".

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter the desired message in the already linked MM channel from the non-connected account.
```
Comment (manojmalik20) [inline data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:38]:
```suggestion
1. Enter the desired message in a linked MM channel from a non-connected account.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Teams_to_MM.md [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Enter any text message, emoji, GiF, link, image, etc. in the already linked channel in Teams or in any DM/GM.
```
Comment (manojmalik20) [inline data/test-cases/plugins/MS Teams sync/Message sync/Teams_to_MM.md:38]:
```suggestion
1. Enter any text message, emoji, GIF, link, image, etc. in a linked channel or any DM/GM in MS Teams.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Message_deleted.md [hunk 16]
```diff
@@ -1,6 +1,6 @@
 ---
 # (Required) Ensure all values are filled up
-name: "Message deleted from Teams to MM"
+name: "Message deleted from MS MS Teams to MM"
```
Comment (manojmalik20) [inline data/test-cases/plugins/MS Teams sync/Message sync/Message_deleted.md:3]:
```suggestion
name: "Message deleted from MS Teams to MM"
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md [hunk 17]
```diff
@@ -35,13 +35,13 @@ steps_hashed: null
 
 **Step 1**
 
-1. Enter the desired message in the already linked MM channel from the non-connected account.
+1. Enter the desired message a already linked MM channel from a non-connected account.
```
Comment (manojmalik20) [inline data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:38]:
```suggestion
1. Enter the desired message in an already linked MM channel from a non-connected account.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md [hunk 18]
```diff
@@ -35,13 +35,13 @@ steps_hashed: null
...
 
 **Step 2**
 
 1. Connect the bot-account.
-2. Enter the desired message in the already linked MM channel by the account which is not already connected.
+2. Enter the desired message a already linked MM channel by the account which is not already connected.
```
Comment (manojmalik20) [inline data/test-cases/plugins/MS Teams sync/Message sync/Non-connected_user.md:43]:
```suggestion
2. Enter the desired message in an already linked MM channel by the account that is not already connected.
```

File: data/test-cases/plugins/MS Teams sync/Message sync/Teams_to_MM.md [hunk 19]
```diff
@@ -35,8 +35,8 @@ steps_hashed: null
 
 **Step 1**
 
-1. Enter any text message, emoji, GiF, link, image, etc. in the already linked channel in Teams or in any DM/GM.
+1. Enter any text message, emoji, GIF, link, image, etc. a already linked channel or in any DM/GM in MS Teams.
```
Comment (manojmalik20) [inline data/test-cases/plugins/MS Teams sync/Message sync/Teams_to_MM.md:38]:
```suggestion
1. Enter any text message, emoji, GIF, link, image, etc. in an already linked channel or in any DM/GM in MS Teams.
```

**Review by manojmalik20 — CHANGES_REQUESTED** [review]

**Pull request merged**